api/handlers.go

```diff
@@ -289,13 +303,42 @@ func (a *API) Unmount(w http.ResponseWriter, r *http.Request) {
 
 	volName := volConfig.String()
 
+	ut := &config.UseMount{
+		Volume:   volName,
+		Reason:   lock.ReasonMount,
+		Hostname: a.Hostname,
+	}
+
+	if !volConfig.Unlocked {
+		a.RemoveStopChan(volName)
+
+		// XXX to doubly ensure we do not UNMOUNT something that is held elsewhere
+		// (presumably because it is mounted THERE instead), we refuse to unmount
+		// anything that doesn't acquire a lock. We also remove the TTL refresh
+		// before taking it so it is not cleared in the unlikely event the mount
+		// takes longer than the TTL. Re-establish the TTL on error only if it is in
+		// locked mode.
+		if err := a.Client.PublishUse(ut); err != nil {
+			a.HTTPError(w, errors.LockFailed.Combine(err))
+			return
+		}
+	}
+
```

Comment:
so there are two kinds of locks in this discussion: permanent and TTL-bound locks.

The first `PublishUse` call takes a permanent lock. It will not expire.

The `startTTLRefresh` will enable a goroutine to periodically refresh the key with a TTL expiry. This ensures that if volplugin or the host crashes, that the mount will eventually expire.